src/MICore/Transports/LocalLinuxTransport.cs

```diff
@@ -187,9 +195,29 @@ public override void InitStreams(LaunchOptions options, out StreamReader reader,
             reader = new StreamReader(gdbStdOutStream);
         }
 
+        private void FifoWatcher_Deleted(object sender, FileSystemEventArgs e)
+        {
+            if (e.FullPath == _gdbStdInName || e.FullPath == _gdbStdOutName)
+            {
+                Logger?.WriteLine("Gdb fifo deleted, stop debugging");
+                _fifoWatcher.Deleted -= FifoWatcher_Deleted;
+                this.Callback.OnDebuggerProcessExit(null);
+            }
+        }
+
         protected override string GetThreadName()
         {
             return "MI.LocalLinuxTransport";
         }
+
+        public override void Close()
+        {
+            base.Close();
```

Comment:
But let's consider this scenario: the user clicks stop debugging and at that very moment, gdb crashes. The file system watcher wakes up and notifies us that the fifos were deleted before the watcher was removed so the code path is taken. Then, the call to stop also enters this code on another thread and also enters this function. Are you sure this can't happen?